### PR TITLE
Migration to remove db_index from sentry_file_timestamp and sentry_file_checksum

### PR DESCRIPTION
**Pull request by iProgramStuff**
These are no longer indices in the database, so following up with a migration.

### CODE REVIEW

**Review by mattrobenolt — CHANGES_REQUESTED** [review]
As I mentioned in the other PR, let's make these a `DROP INDEX CONCURRENTLY IF EXISTS`. And set `atomic = False`.

**Comment by wedamija** [comment]
> As I mentioned in the other PR, let's make these a `DROP INDEX CONCURRENTLY IF EXISTS`. And set `atomic = False`.

This has already been run on Saas by @JTCunning, do we need to do this here for other installations?

File: src/sentry/migrations/0166_drop_file_indices.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    # This flag is used to mark that a migration shouldn't be automatically run in
+    # production. We set this to True for operations that we think are risky and want
+    # someone from ops to run manually and monitor.
+    # General advice is that if in doubt, mark your migration as `is_dangerous`.
+    # Some things you should always mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that
+    #   they can be monitored. Since data migrations will now hold a transaction open
+    #   this is even more important.
+    # - Adding columns to highly active tables, even ones that are NULL.
+    is_dangerous = False
```
Comment (wedamija) [inline src/sentry/migrations/0166_drop_file_indices.py:17]: Mark this as `is_dangerous` so that it doesn't run on saas, since it has already been done.

**Comment by mattrobenolt** [comment]
> This has already been run on Saas by @JTCunning, do we need to do this here for other installations?

Yes please. There's 0 downside to doing this just to be safe. We should just always do this.

**Comment by wedamija** [comment]
> > This has already been run on Saas by @JTCunning, do we need to do this here for other installations?
> 
> Yes please. There's 0 downside to doing this just to be safe. We should just always do this.

Ok. My concern here is that if the indexes were created in south then the name can be different. Hardcoding the names isn't really safe to do anymore. Just looking into how to do this safely with the schema editor.


```

**Review by wedamija — COMMENTED** [review]
@iProgramStuff This should do what you need. It's probably worth generalising this so that we could pass the model name and a list of columns to remove indexes from. Then we can use this for other column index drops.
```
def drop_index(apps, schema_editor):
    File = apps.get_model("sentry", "File")
    for column_name in ["timestamp", "checksum"]:
        index_names = schema_editor._constraint_names(
            File, [column_name], index=True, type_=models.indexes.Index.suffix
        )
        for index_name in index_names:
            schema_editor.execute(f'DROP INDEX CONCURRENTLY IF EXISTS "{index_name}"')
```

File: src/sentry/utils/query.py [hunk 2]
```diff
@@ -237,3 +237,12 @@ def bulk_delete_objects(
         )
 
     return has_more
+
+
+def drop_column_indexes(schema_editor, model, indexes):
```
Comment (wedamija) [inline src/sentry/utils/query.py:242]: Maybe `column_names` instead of indexes? Or `indexed_columns`?

Comment (wedamija) [reply src/sentry/utils/query.py:242]: Also worth having a comment that explains what this does. Something along the lines of:
"Accepts a list of columns and removes any indexes that are only relevant for those columns.". Wording is a bit hard...

**Review by wedamija — COMMENTED** [review]
Looks good to me. I wonder if we should generalize this further and allow pairs of indexes to be passed as well. IE We could pass `[('col_a', 'col_b'), ('col_a', )]` to remove any indexes that are compound on `col_a` and `col_b`, and also any that just use `col_a`. Maybe something we can do as it comes up.

**Pull request closed without merging**